lib/whitehall/document_collection_group_membership_updater.rb

```diff
@@ -0,0 +1,28 @@
+module Whitehall
+  class DocumentCollectionGroupMembershipUpdater
+    attr_reader :document
+
+    def initialize(document)
+      @document = document
+    end
+
+    def call
+      return unless document.published_edition
+
+      document_collection_group_memberships = DocumentCollectionGroupMembership.where(document_id: document.id)
+
+      return unless document_collection_group_memberships.any?
+
+      non_whitehall_link = DocumentCollectionNonWhitehallLink.create(
+        content_id: document.content_id,
+        title: document.published_edition.title,
+        base_path: Whitehall.url_maker.public_document_path(document.published_edition),
+        publishing_app: "content-publisher",
```

Comment:
Should we have this app passed in as a parameter? It feels to be the only place where we refer directly to content publisher and seems a surprising part of this class.